apiserver/facades/controller/charmrevisionupdater/updater_test.go

```diff
@@ -32,36 +33,41 @@ import (
 )
 
 type updaterSuite struct {
-	model        *mocks.MockModel
-	state        *mocks.MockState
-	objectStore  *mocks.MockObjectStore
-	cloudService *commonmocks.MockCloudService
-	resources    *statemocks.MockResources
+	servicefactorytesting.ServiceFactorySuite
+	model              *mocks.MockModel
+	state              *mocks.MockState
+	objectStore        *mocks.MockObjectStore
+	cloudService       *commonmocks.MockCloudService
+	modelConfigService *mocks.MockModelConfigService
+	resources          *statemocks.MockResources
+	client             *mocks.MockCharmhubRefreshClient
 
 	clock clock.Clock
 }
 
 var _ = gc.Suite(&updaterSuite{})
 
-type newCharmhubClientFunc = func(st charmrevisionupdater.State) (charmrevisionupdater.CharmhubRefreshClient, error)
+type newCharmhubClientFunc = func(context.Context) (charmrevisionupdater.CharmhubRefreshClient, error)
```

Comment:
We have been pulling tests back out of `_test` packages and into the package they are testing. You don't have to do as part of this PR but it is a drive by we have been doing when it makes sense. Would allow you to ditch this type.

Reply:
It should be, but it's outside of the scope of this work.